service/src/main/java/bio/terra/catalog/iam/User.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.catalog.iam;
+
+import bio.terra.common.exception.UnauthorizedException;
+import bio.terra.common.iam.BearerTokenParser;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.RequestScope;
+
+/**
+ * This represents the user associated with the current HTTP request. If injected into a singleton
+ * scope class, such as a service, spring will create a proxy object so the value for the current
+ * http request is returned. If no http scope exists when this is used, the proxy will throw an
+ * exception.
+ */
+@Component
+@RequestScope
+public class User {
```

Comment:
Assuming https://github.com/DataBiosphere/terra-data-catalog/pull/133 is merged, this can probably be replaced with `BearerToken`.

Reply:
I've replaced it with `BearerToken` and moved the bean wiring to `BeanConfig`.